# Hypothesis Testing Assignment
## 1. Background & Scenario
Bombay Hospitality Ltd. operates a franchise model for producing exotic Norwegian dinners across New England.
According to their cost model, the weekly operating cost for a franchise is calculated as:
W=₹1000+₹5×XW = ₹1000 + ₹5 \times XW=₹1000+₹5×X 
Where:

•	W = weekly operating cost

•	X = number of units produced per week

For a restaurant producing 600 units per week:
W=₹1000+₹5×600=₹4000W = ₹1000 + ₹5 \times 600 = ₹4000W=₹1000+₹5×600=₹4000 
However, recent feedback from restaurant owners suggests that actual weekly operating costs are higher than this predicted value. To investigate, the company collected data from 25 restaurants, finding that the average weekly cost was ₹3050. The goal is to test the claim that the real mean cost is greater than ₹4000.


# 2. Objective
To perform hypothesis testing using the Z-test to check if the actual mean weekly operating cost is significantly greater than the predicted cost from the company’s cost model.


# 3. Problem Data
Parameter	Value	Description
xˉ\bar{x}xˉ	₹3050	Sample mean weekly cost
μ\muμ	₹4000	Theoretical mean cost from the model
nnn	25	Sample size
σ\sigmaσ	₹125	Standard deviation of weekly cost
α	0.05	Significance level


# 4. Hypotheses
•	Null Hypothesis (H₀): The true mean weekly cost = ₹4000 (model is accurate)
•	Alternative Hypothesis (H₁): The true mean weekly cost > ₹4000 (cost is higher than model predicts)
This is a one-tailed test.


# 5. Step-by-Step Python Implementation

In [4]:

#  1. Import Required Libraries
import numpy as np
from scipy.stats import norm


In [5]:

# 2. Given Data
sample_mean = 3050                # x̄: Sample mean weekly cost
X_mean_units = 600                # Mean units produced per week
model_fixed_cost = 1000           # Fixed cost in Rs.
model_variable_cost_per_unit = 5  # Variable cost per unit in Rs.
unit_sd = 25                      # Standard deviation of units produced
n = 25                            # Sample size


In [6]:

# 3. Calculate Theoretical Mean Cost (μ) from cost model
# W = Fixed Cost + (Variable Cost * Mean Units)
mu = model_fixed_cost + (model_variable_cost_per_unit * X_mean_units)


In [7]:
# 4. Calculate Standard Deviation of Weekly Costs (σ)
# σ = variable cost per unit * standard deviation of units produced
sigma = model_variable_cost_per_unit * unit_sd


In [8]:

# 5. Calculate Z Test Statistic
z_stat = (sample_mean - mu) / (sigma / np.sqrt(n))


In [9]:

# 6. Determine Critical Value at α = 0.05 (One-tailed test)
alpha = 0.05
critical_value = norm.ppf(1 - alpha)


In [10]:

# 7. State Hypotheses
H0 = f"H₀: The true mean weekly cost = Rs.{mu} (model is accurate)"
H1 = f"H₁: The true mean weekly cost > Rs.{mu} (cost is higher than model)"


In [11]:

# 8. Print Results
print("--- Given Data ---")
print(f"Sample Mean (x̄): Rs.{sample_mean}")
print(f"Theoretical Mean (μ): Rs.{mu}")
print(f"Sample Size (n): {n}")
print(f"Standard Deviation of Weekly Costs (σ): Rs.{sigma}")

print("\n--- Hypotheses ---")
print(H0)
print(H1)

print("\n--- Test Statistic & Critical Value ---")
print(f"Z-Statistic: {z_stat:.4f}")
print(f"Critical Value (α = 0.05): {critical_value:.4f}")


--- Given Data ---
Sample Mean (x̄): Rs.3050
Theoretical Mean (μ): Rs.4000
Sample Size (n): 25
Standard Deviation of Weekly Costs (σ): Rs.125

--- Hypotheses ---
H₀: The true mean weekly cost = Rs.4000 (model is accurate)
H₁: The true mean weekly cost > Rs.4000 (cost is higher than model)

--- Test Statistic & Critical Value ---
Z-Statistic: -38.0000
Critical Value (α = 0.05): 1.6449


In [12]:

# 9. Decision
if z_stat > critical_value:
    print("\nDecision: Reject H₀ → There is strong evidence that weekly costs are higher than the model predicts.")
else:
    print("\nDecision: Fail to Reject H₀ → No strong evidence that weekly costs are higher than the model predicts.")



Decision: Fail to Reject H₀ → No strong evidence that weekly costs are higher than the model predicts.


# 7. Interpretation
•	Z-statistic = -38.00 → This is far below the critical value (1.6449).

•	Since our sample mean (₹3050) is much less than the predicted ₹4000, the result is in the opposite direction of the claim.

•	Therefore, we fail to reject H₀.


# 8. Conclusion
Based on our hypothesis test, there is no statistical evidence to support the restaurant owners’ claim that weekly operating costs are higher than the company’s cost model predicts.
In fact, the data shows that the observed average cost is lower than predicted.
